### ISSUE TITLE
SRResNet records a padding hparam that its upsample convs ignore

### ISSUE THREAD

**Issue by YeapJieShen** [YeapJieShen/single-image-super-resolution#241]
## The defect

`SRResNet.__init__` takes `padding` and threads it into the head, the residual blocks and the
tail. `SRUpsampleBlock` hardcodes it:

```python
torch.nn.Conv2d(channels, channels * (scale**2), kernel_size=kernel_size, padding="same")
```

Measured on `SRResNet(scale=2, padding=0)`:

| padding in effect | conv count | example |
|---|---:|---|
| `(0, 0)` — as configured | **35** | `head.0` |
| `same` — ignored the argument | **1** | `upsample.0.upsample.0` |

## Why it is worth fixing even though nothing hits it today

Every shipped config uses `padding: same`, so the inconsistency is currently invisible. Two
reasons it still matters:

1. **The recorded hyperparameter becomes a lie.** `padding` goes into `_hparams`, and `_hparams`
   goes verbatim into checkpoint and ONNX provenance as `model.init_args`. A file would claim
   `padding: 0` while one of its convolutions was built `same`. This project treats artifact
   metadata as the thing a downstream consumer trusts to know what a file means.
2. **It is silent.** Nothing raises, nothing warns; the model simply mixes two padding regimes.
   The output shape still works out, so the first sign would be an unexplained metric.

Contrast `SRCNN`, where `padding` reaches every convolution, and where a mode that could not be
expressed (`eval_padding='same'` with an even kernel) raises rather than silently doing something
else.

## The judgement call the fix needs

`SRUpsampleBlock` feeds `PixelShuffle`, which needs its input to be exactly `channels * scale**2`
at the same spatial size — so a valid convolution there shrinks the feature map before the shuffle
and changes the output geometry. It may be that `same` is genuinely required here rather than
merely convenient.

If so, **the fix is to say that, not to thread the argument through**: `SRUpsampleBlock` should
document why it is fixed, and `SRResNet` should either reject a non-`same` `padding` outright or
record what was actually built. What it must not do is accept an argument, record it, and ignore it.

## Acceptance

- [ ] `padding` is either honoured everywhere, or rejected/normalised where it cannot be
- [ ] `_hparams["padding"]` describes what was actually constructed
- [ ] A test asserting every `Conv2d` in the built model agrees with the recorded hparam
